azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java

```diff
@@ -75,17 +87,33 @@ public ReadablePeriod getPeriod() {
   public long getNextCheckTime() {
     return nextCheckTime;
   }
+  public String getCronExpression() {
+    return cronExpression;
+  }
+
+  private void parseCronExpression() {
+    if (cronExpression != null && timezone != null) {
+      try {
+        cronExecutionTime = new CronExpression(cronExpression);
+        cronExecutionTime.setTimeZone(TimeZone.getTimeZone(timezone.getID()));
+      } catch (ParseException pe) {
+        logger.error("This expression " + cronExpression + " can not be parsed to quartz cron.");
```

Comment:
Actually, this should not happen. As I already check the legality of cron Expression in Schedule Servlet.